diff --git a/functions/_fzf_configure_bindings_help.fish b/functions/_fzf_configure_bindings_help.fish
index ecfe68e..a1856b3 100644
--- a/functions/_fzf_configure_bindings_help.fish
+++ b/functions/_fzf_configure_bindings_help.fish
@@ -19,0 +20 @@ DESCRIPTION
+        Search Files       |  Ctrl+Q     (Q for ???...)    |  --files
diff --git a/functions/_fzf_preview_file.fish b/functions/_fzf_preview_file.fish
index eaa68d0..ecadfc3 100644
--- a/functions/_fzf_preview_file.fish
+++ b/functions/_fzf_preview_file.fish
@@ -5 +5,5 @@ function _fzf_preview_file --description "Print a preview for the given file bas
-    set file_path $argv
+    set file_path $argv[1]
+    if test (count $argv) -gt 1
+      set line_number $argv[2]
+      set bat_highlight_opt --highlight-line $line_number
+    end
@@ -21 +25 @@ function _fzf_preview_file --description "Print a preview for the given file bas
-            bat --style=numbers --color=always "$file_path"
+            bat --style=numbers --color=always $bat_highlight_opt "$file_path"
diff --git a/functions/_fzf_search_processes.fish b/functions/_fzf_search_processes.fish
index 9424aad..dda66b5 100644
--- a/functions/_fzf_search_processes.fish
+++ b/functions/_fzf_search_processes.fish
@@ -4,0 +5,5 @@ function _fzf_search_processes --description "Search all running processes. Repl
+    if not set --query fzf_ps_opts
+      set ps_opts '-A -opid,command'
+    else
+      set ps_opts $fzf_ps_opts
+    end
@@ -6 +11 @@ function _fzf_search_processes --description "Search all running processes. Repl
-        ps -A -opid,command | \
+        ps $ps_opts | \
diff --git a/functions/fzf_configure_bindings.fish b/functions/fzf_configure_bindings.fish
index ec24f73..0ab9e55 100644
--- a/functions/fzf_configure_bindings.fish
+++ b/functions/fzf_configure_bindings.fish
@@ -7 +7 @@ function fzf_configure_bindings --description "Installs the default key bindings
-    set options_spec h/help 'directory=?' 'git_log=?' 'git_status=?' 'history=?' 'processes=?' 'variables=?'
+    set options_spec h/help 'directory=?' 'git_log=?' 'git_status=?' 'history=?' 'processes=?' 'variables=?' 'files=?'
@@ -19 +19 @@ function fzf_configure_bindings --description "Installs the default key bindings
-        set key_sequences \e\cf \e\cl \e\cs \cr \e\cp \cv # \c = control, \e = escape
+        set key_sequences \e\cf \e\cl \e\cs \cr \e\cp \cv \cq # \c = control, \e = escape
@@ -25,0 +26 @@ function fzf_configure_bindings --description "Installs the default key bindings
+        set --query _flag_files && set key_sequences[7] "$_flag_files"
@@ -38,0 +40 @@ function fzf_configure_bindings --description "Installs the default key bindings
+            test -n $key_sequences[7] && bind --mode $mode $key_sequences[7] _fzf_search_files
